kernel/source/filedesc/calls_inner.c

```diff
@@ -61,6 +61,42 @@ static s32 GetThreadSpecificMsgOrFreeFromExtra(const int useMsgFromExtraInsteadO
 	return IPC_EMAX;
 }
 
+static bool IOSFD_CheckIdValidForProcess(const s32 fd)
+{
+#ifdef MIOS
+	if (fd >= MAX_PROCESS_FDS || ProcessFileDescriptors[GetProcessID()][fd].BelongsToResource == NULL)
+		return false;
+#else
+	if (!(0x10000 <= fd && fd < 0x10002))
+	{
+		if (fd >= MAX_PROCESS_FDS || ProcessFileDescriptors[CurrentThread == IpcHandlerThread ? 15 : GetProcessID()][fd].BelongsToResource == NULL)
+			return false;
+	}
+#endif
+
+	return true;
+}
+
+static FileDescriptor* IOSFD_GetProcessFdFromId(const s32 fd)
```

Comment:
`GetProcessFileDescriptor` or something, with parameter `id` or something. im having some trouble understanding what the function does from just the name

Reply:
I'm using Fd/FD/File Descriptor here to refer to the actual type/pointer of this type, and fd/Id/id of type 'int' or such to refer to the more common 'handle' part, which is an index (id) into the file descriptor table for the process (which contains the values with the meaning I described first).

For example on linux, 'fd 1' (aka stdout) is an int (handle/index) that refers to the file descriptor (struct with data for accessing files) at position 1 in the kernel's table/array for the current process (0-indexed).

I guess fh 'file handle' being an index into the fd 'file descriptor' array (that means the actual file) would be simpler?

Reply:
maybe just renaming the `fd` argument to `id` and removing the `FromId` will do then. like i said on discord its a bit of a complex naming issue and as developers naming isn't our strong suit haha

what you said is right, i know that but its also a complex mess atm hehe